code/game/objects/items/weapons/weldbackpack.dm

```diff
@@ -1,85 +1,209 @@
+	tank         = null
 
+	var/obj/item/chems/weldpack/linked_pack
+
+/obj/item/weldingtool/weldpack/Initialize(ml, material_key, var/obj/item/chems/weldpack/pack)
+	. = ..()
+	if(pack)
+		linked_pack = pack
+	if(linked_pack)
+		linked_pack.register_welder_callbacks(src, /obj/item/weldingtool/weldpack/proc/on_pack_dropped, /obj/item/weldingtool/weldpack/proc/on_pack_deleted)
+
+/obj/item/weldingtool/weldpack/insert_tank(obj/item/chems/welder_tank/T, mob/user, no_updates, quiet)
+	return FALSE
+
+/obj/item/weldingtool/weldpack/remove_tank(mob/user)
+	return FALSE
+
+/obj/item/weldingtool/weldpack/toggle_unscrewed(mob/user)
+	return FALSE
+
+/obj/item/weldingtool/weldpack/attempt_modify(obj/item/W, mob/user)
+	return FALSE
+
+/obj/item/weldingtool/weldpack/dropped(mob/user)
+	. = ..()
+	if(linked_pack)
+		linked_pack.reattach_gun(user)
+
+/obj/item/weldingtool/weldpack/get_fuel()
+	return linked_pack? linked_pack.get_fuel() : 0
+
+/obj/item/weldingtool/weldpack/use_fuel(amount)
+	. = TRUE
+	if(get_fuel() < amount)
+		. = FALSE //Try to burn as much as possible anyways
+	if(linked_pack)
+		linked_pack.reagents.remove_reagent(/decl/material/liquid/fuel, amount)
+
+/**Called by the parent when the welderpack is dropped */
+/obj/item/weldingtool/weldpack/proc/on_pack_dropped(var/mob/user)
+	if(!linked_pack.is_welder_attached())
+		linked_pack.reattach_gun(user)
+
+/**Called by the parent when the welderpack is deleting */
+/obj/item/weldingtool/weldpack/proc/on_pack_deleted()
+	if(!linked_pack.is_welder_attached())
+		linked_pack.reattach_gun()
+
+/obj/item/weldingtool/weldpack/get_storage_cost()
+	if(loc != linked_pack)
+		return ITEM_SIZE_NO_CONTAINER
+	return ..()
+
+////////////////////////////////////////////////////////////
+//Welder Pack
+////////////////////////////////////////////////////////////
+/obj/item/chems/weldpack
+	name              = "welding kit"
+	desc              = "An unwieldy, heavy backpack with two massive fuel tanks. Includes a connector for most models of portable welding tools."
+	icon              = 'icons/obj/items/welderpack.dmi'
+	icon_state        = ICON_STATE_WORLD
+	slot_flags        = SLOT_BACK
+	w_class           = ITEM_SIZE_HUGE
+	volume            = 350
+	var/obj/item/weldingtool/weldpack/welder
+	var/datum/callback/call_on_pack_dropped
+	var/datum/callback/call_on_pack_deleting
+
+/obj/item/chems/weldpack/populate_reagents()
+	reagents.add_reagent(/decl/material/liquid/fuel, reagents.maximum_volume)
+
+/obj/item/chems/weldpack/Initialize()
 	. = ..()
+	if(!welder)
+		welder = new(src, null, src)
 
-/obj/item/weldpack/Destroy()
+/obj/item/chems/weldpack/Destroy()
+	if(call_on_pack_deleting)
+		call_on_pack_deleting.Invoke()
+	QDEL_NULL(call_on_pack_deleting)
+	QDEL_NULL(call_on_pack_dropped)
 	QDEL_NULL(welder)
 
 	. = ..()
 
-/obj/item/weldpack/attackby(obj/item/W, mob/user)
+/obj/item/chems/weldpack/proc/register_welder_callbacks(var/obj/item/weldingtool/weldpack/W, var/ondropped, var/ondelete)
```

Comment:
I'm not sure why you're saying that's over-engineered? 
If you drop the welder pack, the welder gun has to drop as well? Otherwise you can drop the welder pack and just walk with the welder around as if you were still wearing the pack?

Reply:
Alright, yeah I see what you mean now. Honestly, I'm not entirely sure.

Reply:
Will take a look. I'll have to test without the callbacks. Its been a while since I wrote that, so I don't really recall why..

Reply:
Yeah, it seems to work fine without. So went with that.